staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config_test.go

```diff
@@ -507,3 +510,185 @@ resources:
 		})
 	}
 }
+
+func TestIsKMSPluginReady(t *testing.T) {
+	var testCases = []struct {
+		desc       string
+		status     int
+		healthzURL string
+		want       bool
+	}{
+		{
+			desc:   "Healthz OK",
+			status: http.StatusOK,
+			want:   true,
+		},
+		{
+			desc:   "Healthz !OK",
+			status: http.StatusServiceUnavailable,
+		},
+		{
+			desc:       "Healthz server down",
+			status:     http.StatusServiceUnavailable,
+			healthzURL: "http://dead-url:8080/healthz",
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.desc, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(tt.status)
+			}))
+			defer ts.Close()
+
+			config := &apiserverconfig.KMSConfiguration{
+				HealthzURL: ts.URL,
+				HealthzTimeout: &metav1.Duration{
+					Duration: 1 * time.Second,
+				},
+			}
+
+			if tt.healthzURL != "" {
+				config.HealthzURL = tt.healthzURL
+			}
+
+			r, _ := isKMSPluginReady(config)
+			if r != tt.want {
+				t.Errorf("got %v want %v for healthz status", r, tt.want)
+			}
+
+		})
+	}
+}
+
+func TestGetHealthz(t *testing.T) {
```

Comment:
This test seems redundant with `TestIsKMSPluginReady`. It doesn't seem to cover any new code paths